dbms/src/Interpreters/ExternalLoader.cpp

```diff
@@ -528,20 +530,22 @@ class ExternalLoader::LoadingDispatcher : private boost::noncopyable
     /// The function doesn't touch the objects which were never tried to load.
     void reloadOutdated()
     {
+        /// Iterate through all the objects and find loaded ones which should be checked if they were modified.
         std::unordered_map<LoadablePtr, bool> is_modified_map;
-
         {
             std::lock_guard lock{mutex};
             TimePoint now = std::chrono::system_clock::now();
             for (const auto & name_and_info : infos)
             {
                 const auto & info = name_and_info.second;
-                if ((now >= info.next_update_time) && !info.loading() && info.was_loading())
```

Comment:
This line caused the problem. Because if we tried to load a dictionary before but never succeeded then `info.was_loading() == true`, `info.object == nullptr`, and the function `is_object_modified()` were called with `nullptr` as an argument and we got segfault. Now I've changed the condition in order to call the function `is_object_modified()` only for loaded objects (which are not null). Anyway it doesn't make sense to check for modification not loaded dictionary.